courses/views.py

```diff
@@ -38,6 +41,46 @@ class ProgramViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = ProgramSerializer
 
 
+class ProgramLearnersView(APIView):
+    """API for Learners enrolled in the Program"""
+
+    authentication_classes = (
+        SessionAuthentication,
+        TokenAuthentication,
+    )
+    permission_classes = (
+        IsAuthenticated,
+    )
+    serializer_class = ProfileImageSerializer
+
+    def get(self, request, *args, **kargs):
+        """
+        Get eight random learners with images and
+        the total count of visible learners in the program
+        """
+        program_id = self.kwargs["program_id"]
+        users = ProgramEnrollment.objects.filter(
+            program_id=program_id
+        ).values_list('user', flat=True)
+
+        queryset = Profile.objects.exclude(
+            image_small__exact=''
+        ).filter(user__in=users).exclude(
+            account_privacy='private'
+        ).exclude(
+            user=request.user
+        ).order_by('?')
```

Comment:
Yeah, if you see `values_list` being used there's usually some evaluation happening which is subtle, for example during iteration or `list(...)`: https://docs.djangoproject.com/en/1.11/ref/models/querysets/#when-querysets-are-evaluated

It's also confusing since `repr(...)` will also do it, which means just looking at a queryset in your console will evaluate it